RequestKit/Router.swift

```diff
@@ -43,7 +43,12 @@ public protocol Router {
 public extension Router {
     public func request() -> NSURLRequest? {
         let URLString = configuration.apiEndpoint.stringByAppendingURLPath(path)
-        var parameters = encoding == .JSON ? [:] : params
+        var parameters: [String: String] = [:]
+        if encoding != .JSON {
```

Comment:
Ok, I investigated this PR locally a bit further. Your tests are ok for JSON encoding and you are right about them being superfluous because you basically test `NSJSONSerialisation`.
However, this PR does not change the JSON implementation but the URL Encoding implementation. Testing this has the following result:

Given the params: `["key1": "value1", "key2": "value2", "bool": true, "array": ["one", "two"]]`
Your implementation has the following result: `https://example.com/api/v1/some_route?access_token=1234&array=%28%0A%20%20%20%20one%2C%0A%20%20%20%20two%0A%29&bool=1&key1=value1&key2=value2`
Which translates to:
```
https://example.com/api/v1/some_route?access_token=1234&array=(
    one,
    two
)&bool=1&key1=value1&key2=value2
```
But it should be: `https://example.com/api/v1/some_route?access_token=1234&array[]=one&array[]=two&bool=1&key1=value1&key2=value2&key=true`

The essential problem is that you take the `description` of the `Array` and put it into a URL which has not the same result as implementing the encoding logic. Accepting only `[String: String]` is necessary because nothing else is implemented currently. Accepting `[String: AnyObject]` might work for JSON encoding but users of the library might feel inclined to think that this also works for `URL` encoding, which it does not.